### PR TITLE
 Added idField to the resource during the update operation.

### PR DESCRIPTION
**Pull request by andriikonstantinov**
The id field wasn't passed to the extension during update events.
Added new test verifying pre_update and pre_update_in_transaction id changes.

### CODE REVIEW

**Comment by andriikonstantinov** [comment]
Supersedes #485

**Comment by morrisson** [comment]
Isn't context.id enough?

**Comment by p-kozlowski** [comment]
@morrisson IMO it's a little obscure to read all the resource data from "resource" field and id from "context". Let's keep the consistency and Principle Of Least Astonishment.

**Comment by morrisson** [comment]
Then you should do the same thing for pre_delete.

**Comment by przemyslaw-dobrowolski-cl** [comment]
@andriikonstantinov please rebase

File: server/resources/resource_management.go [hunk 1]
```diff
@@ -611,13 +611,21 @@ func UpdateResource(
 	if err != nil {
 		return ResourceError{err, err.Error(), Unauthorized}
 	}
+	whetherToDeleteId := false
```
Comment (nati) [inline server/resources/resource_management.go]: nit: whetherToDeleteID
plz use go lint

Comment (nati) [inline server/resources/resource_management.go]: I prefer "needDeleted"

Comment (andriikonstantinov) [inline server/resources/resource_management.go]: Sure.Done

File: server/resources/resource_management.go [hunk 2]
```diff
@@ -627,13 +627,21 @@ func UpdateResource(
 	if err != nil {
 		return ResourceError{err, err.Error(), Unauthorized}
 	}
+	needDeleted := false
```
Comment (przemyslaw-dobrowolski-cl) [inline server/resources/resource_management.go]: Maybe it meant to be needDelete?

**Comment by andriikonstantinov** [comment]
@nati @przemyslaw-dobrowolski-cl  please review

**Pull request merged**